File: server/src/main/java/com/fluffy/reaction/domain/ReactionRepository.java
package com.fluffy.reaction.domain;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ReactionRepository extends JpaRepository<Reaction, Long> {

    Optional<Reaction> findByTargetTypeAndTargetIdAndMemberIdAndType(
            String targetType,
            Long targetId,
            Long memberId,
            ReactionType type
    );
}

File: server/src/main/kotlin/com/fluffy/reaction/domain/ReactionRepository.kt
package com.fluffy.reaction.domain

import org.springframework.data.jpa.repository.JpaRepository

interface ReactionRepository : JpaRepository<Reaction, Long> {

    fun findByTargetTypeAndTargetIdAndMemberIdAndType(
        targetType: String,
        targetId: Long,
        memberId: Long,
        type: ReactionType
    ): Reaction?
}